diff --git a/src/or/entrynodes.c b/src/or/entrynodes.c
index 70f7c87..e5aa74c 100644
--- a/src/or/entrynodes.c
+++ b/src/or/entrynodes.c
@@ -1630,6 +1630,10 @@ get_configured_bridge_by_addr_port_digest(const tor_addr_t *addr,
           !tor_addr_compare(&bridge->addr, addr, CMP_EXACT) &&
           bridge->port == port)
         return bridge;
+      if (!digest &&
+          !tor_addr_compare(&bridge->addr, addr, CMP_EXACT) &&
+          bridge->port == port)
+        return bridge;
       if (digest && tor_memeq(bridge->identity, digest, DIGEST_LEN))
         return bridge;
     }
